src/Okta.Sdk.IntegrationTests/NetworkZoneScenarios.cs

```diff
@@ -0,0 +1,134 @@
+﻿// <copyright file="NetworkZoneScenarios.cs" company="Okta, Inc">
+// Copyright (c) 2020 - present Okta, Inc. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Xunit;
+
+namespace Okta.Sdk.IntegrationTests
+{
+    public class NetworkZoneScenarios
+    {
+        [Fact]
+        public async Task CreateUpdateDeleteNetworkZone()
+        {
+            var oktaClient = TestClient.Create();
+            var zonesClient = oktaClient.NetworkZones;
+            NetworkZone newZone = BuildNetworkZoneObject();
+            var createZoneResponse = await zonesClient.CreateNetworkZoneAsync(newZone);
+            try
+            {
+                createZoneResponse.Id.Should().NotBeNullOrEmpty();
+                createZoneResponse.Name.Should().Be(newZone.Name);
+                createZoneResponse.Type.Should().Be(newZone.Type);
+                createZoneResponse.Status.Should().Be(newZone.Status);
+                createZoneResponse.Gateways.Should().HaveCount(2);
+                createZoneResponse.Proxies.Should().HaveCount(2);
+
+                createZoneResponse.Name = "Another Name";
+                createZoneResponse.Gateways.Add(new NetworkZoneAddress
+                {
+                    Type = NetworkZoneAddressType.Cidr,
+                    Value = "1.1.1.1/24",
+                });
+
+                var updateZoneResponse = await zonesClient.UpdateNetworkZoneAsync(createZoneResponse, createZoneResponse.Id);
+                updateZoneResponse.Name.Should().Be(createZoneResponse.Name);
+                updateZoneResponse.Gateways.Count.Should().Be(3);
+            }
+            finally
+            {
+                await zonesClient.DeleteNetworkZoneAsync(createZoneResponse.Id);
```

Comment:
I'm not personally a fan of testing multiple different operations in a single unit test.  If any of the operations create, update or delete fail it will not be immediately obvious which operation needs attention.  Is it possible to specify failure messages that expose which operation failed if any?

Reply:
It's a cleanup code that cannot be separated from the other. I need this `delete` to be called after `create` in this order.